tests/unit/aiplatform/test_models.py

```diff
@@ -0,0 +1,362 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import pytest
+import importlib
+from unittest import mock
+
+from google.api_core import operation as ga_operation
+import google.auth
+from google.auth import credentials as auth_credentials
+from google.cloud import aiplatform
+from google.cloud.aiplatform import initializer
+from google.cloud.aiplatform import models
+from google.cloud.aiplatform import utils
+from google.cloud.aiplatform_v1beta1.services.model_service.client import (
+    ModelServiceClient,
+)
+from google.cloud.aiplatform_v1beta1.types import env_var
+from google.cloud.aiplatform_v1beta1.types import model as gca_model
+from google.cloud.aiplatform_v1beta1.types import model_service
+
+_TEST_PROJECT = "test-project"
+_TEST_PROJECT_2 = "test-project-2"
+_TEST_LOCATION = "us-central1"
+_TEST_LOCATION_2 = "us-central2"
+_TEST_MODEL_NAME = "test-model"
+_TEST_ARTIFACT_URI = "gs://test/artifact/uri"
+_TEST_SERVING_CONTAINER_IMAGE = "gcr.io/test-serving/container:image"
+_TEST_SERVING_CONTAINER_PREDICTION_ROUTE = "predict"
+_TEST_SERVING_CONTAINER_HEALTH_ROUTE = "metadata"
+_TEST_DESCRIPTION = "test description"
+_TEST_SERVING_CONTAINER_COMMAND = ["python3", "run_my_model.py"]
+_TEST_SERVING_CONTAINER_ARGS = ["--test", "arg"]
+_TEST_SERVING_CONTAINER_ENVIRONMENT_VARIABLES = {
+    "learning_rate": 0.01,
+    "loss_fn": "mse",
+}
+_TEST_SERVING_CONTAINER_PORTS = [8888, 10000]
+_TEST_MODEL_ID = 1234
+
+
+class TestModel:
```

Comment:
Yes, https://docs.pytest.org/en/stable/getting-started.html#group-multiple-tests-in-a-class

```
pytest discovers all tests following its Conventions for Python test discovery, so it finds both test_ prefixed functions. There is no need to subclass anything, but make sure to prefix your class with Test otherwise the class will be skipped. We can simply run the module by passing its filename:```